ui/src/lib/fiber.ts

```diff
@@ -0,0 +1,53 @@
+// Define the type signature of an Action: async function or Promise
+export type Action = () => Promise<void>
+
+/**
+ * This is a simple mechanism to ensure that we execute async or promise
+ * returning code in the order it was added to the queue.
+ */
+export class Fiber {
+
+  /**
+   * Define the "fiber" that we'll use. For a reminder: Process > Thread > Fiber
+   */
+  private actionFiber: Promise<void>;
+
+  private actionQueue: Action[];
+
+  /**
+   * Create the first resolved promise from which we build
+   */
+  constructor() {
+    this.actionFiber = Promise.resolve();
+  }
+
+  /**
+   * This takes an array of Promises or async functions (thunks) and ensures they
+   * are executed in sequential fashion over the lifetime of this fiber.
+   *
+   * **Note:** Do not enclose any React state variables or function parameters
```

Comment:
Might be good to have an example of what this would look like in the comments here.